graphql-dgs-spring-boot-micrometer/src/main/kotlin/com/netflix/graphql/dgs/metrics/micrometer/dataloader/BatchLoaderWithContextInterceptor.kt

```diff
@@ -51,9 +52,15 @@ internal class BatchLoaderWithContextInterceptor(
                     )
                 }
             } catch (exception: Exception) {
-                logger.warn("Error creating timer interceptor '{}' for {} with exception {}", ID, javaClass.simpleName, exception.message)
-                @Suppress("UNCHECKED_CAST")
-                method.invoke(batchLoaderWithContext, *(args)) as CompletionStage<List<*>>
+                //If the dataloader itself fails just propagate the exception
+                if(exception is InvocationTargetException) {
+                    throw exception.targetException
+                } else {
+                    logger.warn("Error creating timer interceptor '{}' for {} with exception {}", ID, javaClass.simpleName, exception.message)
+                    @Suppress("UNCHECKED_CAST")
+                    method.invoke(batchLoaderWithContext, *(args)) as CompletionStage<List<*>>
+                }
+
             }
```

Comment:
imho catching each exception type is cleaner:
```
            } catch (exception: InvocationTargetException) {
                //If the dataloader itself fails just propagate the exception
                throw exception.targetException
            } catch (exception: Exception) {
                logger.warn("Error creating timer interceptor '{}' for {} with exception {}", ID, javaClass.simpleName, exception.message)
                @Suppress("UNCHECKED_CAST")
                method.invoke(batchLoaderWithContext, *(args)) as CompletionStage<List<*>>
            }
```

Reply:
Of course, I don't know why I didn't do that to begin with 🤦‍♂️